### PR TITLE
TF-2189 Fix sending queue on iOS

### PR DESCRIPTION
**Pull request by dab246**
## Issue

- #2189 

## Dependent 

- Need merged: https://github.com/linagora/enough_html_editor/pull/23

## Resolved

https://github.com/linagora/tmail-flutter/assets/80730648/b9a3662b-b035-4c36-a083-5ff056e14964

### CODE REVIEW

File: lib/features/composer/presentation/composer_controller.dart [hunk 1]
```diff
@@ -862,134 +865,128 @@ class ComposerController extends BaseController {
...
+    if (composerArguments.value == null ||
+        mailboxDashBoardController.userProfile.value == null ||
+        mailboxDashBoardController.sessionCurrent == null ||
+        mailboxDashBoardController.accountId.value == null
+    ) {
+      logError('ComposerController::isParamUserNotNull: Param is NULL');
+      return true;
+    }
+    return false;
+  }
```
Comment (hoangdat) [inline lib/features/composer/presentation/composer_controller.dart:878]:
```suggestion
  void _validateUserParams {
    if (composerArguments.value == null ||
        mailboxDashBoardController.userProfile.value == null ||
        mailboxDashBoardController.sessionCurrent == null ||
        mailboxDashBoardController.accountId.value == null
    ) {
      throw `ABC Exception`;
    }
    return;
  }
```
then handle exception in `final`

Comment (dab246) [reply lib/features/composer/presentation/composer_controller.dart:878]: We don't take any action on it. IMO, it is not necessary to do so on the presentation layer

File: lib/features/composer/presentation/composer_controller.dart [hunk 2]
```diff
@@ -1294,21 +1269,14 @@ class ComposerController extends BaseController {
...
-    if (event.emailIdEditing == null) {
-      mailboxDashBoardController.saveEmailToDraft(
-        session: event.session,
-        accountId: event.accountId,
-        email: newEmail
-      );
-    } else {
-      mailboxDashBoardController.updateEmailToDraft(
+    mailboxDashBoardController.saveEmailToDraft(
+      arguments: SaveToDraftArguments(
```
Comment (hoangdat) [inline lib/features/composer/presentation/composer_controller.dart:1281]: why use `save to draft` for this case? it is also save in local db?

File: pubspec.yaml [hunk 3]
```diff
@@ -239,6 +239,11 @@ dependency_overrides:
 
   pointer_interceptor: 0.9.1
 
+  enough_html_editor:
+    git:
+      url: https://github.com/linagora/enough_html_editor.git
+      ref: improvement/support-get-content-signature
```
Comment (hoangdat) [inline pubspec.yaml:245]: please use `email_supported`

**Review by hoangdat — COMMENTED** [review]
only one question: why we need to change `save to draft` argument?

**Comment by dab246** [comment]
> only one question: why we need to change `save to draft` argument?

We will handle it on the dashboard controller, avoiding the case of the composer controller being canceled

**Review by hoangdat — APPROVED** [review]

**Pull request merged**